### PR TITLE
geogebra: init at 5.0.264.0

### PR DESCRIPTION
**Pull request by moyamo**
###### Motivation for this change

Geogebra package was not included in the repo.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is my first nix package so you should definitely look over it.
# Licensing

I'm not sure whether [the Geogebra License](https://www.geogebra.org/license/) qualifies as a free license. The code is GPL but the assets to not allow commercial use. You may want to look at this more closely.

### CODE REVIEW

File: pkgs/applications/science/math/geogebra/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+let
+  version = "5.0.264.0";
+in
+  stdenv.mkDerivation {
+    name = "geogebra-${version}";
+    inherit version;
+
+    builder = ./builder.sh;
+
+        src = fetchurl {
```
Comment (FRidh) [inline pkgs/applications/science/math/geogebra/default.nix]: please fix indentation

**Comment by FRidh** [comment]
Thanks for the PR @moyamo. You've defined a custom builder to specify the installPhase, however, you can do this is directly in `stdenv.mkDerivation { }`.

**Comment by moyamo** [comment]
Another thing. I'm not sure what is nixpkgs policy on .desktop files. You may want to add a .desktop file to this function.

File: pkgs/applications/science/math/geogebra/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      EOF
+
+      chmod +x "$out/bin/geogebra"
+      '';
+
+    meta = {
+      description = "Dynamic mathematics software with graphics, algebra and spreadsheets";
+      longDescription = ''
+      Dynamic mathematics software for all levels of education that brings
+      together geometry, algebra, spreadsheets, graphing, statistics and
```
Comment (FRidh) [inline pkgs/applications/science/math/geogebra/default.nix]: indentation here

**Comment by FRidh** [comment]
> Another thing. I'm not sure what is nixpkgs policy on .desktop files. You may want to add a .desktop file to this function.

I'm not familiar with .desktop files. @obadz , you are, right?

**Comment by moyamo** [comment]
Sorry, I can't see what's wrong with the indentation. Is there a style guide somewhere? I can't seem to find it.

**Comment by rnhmjoj** [comment]
According to the GNU project a license which does not permit commercial use of the program is classified as "nonfree", [this one](https://www.gnu.org/licenses/license-list.html#UtahPublicLicense) for example. Also the [free software definition](https://www.gnu.org/philosophy/free-sw.html) specifically says " A free program must be available for commercial use, commercial development, and commercial distribution. "

**Comment by FRidh** [comment]
It seems to be GPL3, CC BY-NC-SA 3.0 and GeoGebra, with the last one being unfree. You can add this last license to `lib.licenses`, and then add all three to `meta.licenses`.

**Comment by obadz** [comment]
Pls see https://github.com/NixOS/nixpkgs/blob/3a75ed446410f84829aab648d38fda55f514e523/pkgs/applications/networking/remote/citrix-receiver/default.nix#L84 as an example on how to use `makeDesktopItem`

**Comment by moyamo** [comment]
Should I set `preferLocalBuild` since this function just downloads and extracts a tarball?

File: pkgs/applications/science/math/geogebra/default.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    mimeType = "application/vnd.geogebra.file;application/vnd.geogebra.tool;";
+  };
+
+  installPhase = ''
+    install -dm755 "$out/share/geogebra"
+    install "geogebra/"* -t "$out/share/geogebra/"
+    mkdir "$out/bin"
+
+    cat <<EOF >"$out/bin/geogebra"
+    #! $SHELL
```
Comment (obadz) [inline pkgs/applications/science/math/geogebra/default.nix]: I would suggest using `makeWrapper/wrapProgram` here

**Pull request closed without merging**